### ISSUE TITLE
MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 2 SIGWINCH listeners added to [process].

### ISSUE THREAD

**Issue by timocov** [ai/size-limit#184]
I'm trying to integrate `size-limit` to https://github.com/tradingview/lightweight-charts and run the following script:

```javascript
const sizeLimit = require('size-limit');
const filePlugin = require('@size-limit/file');

// here could be any file instead of dist/lightweight-charts.esm.production.js, e.g. package.json
sizeLimit([filePlugin], ['dist/lightweight-charts.esm.production.js']).then(result => {
	console.log(result);
});
```

I use NodeJS 14.4 if it's related.

If I run with `--trace-warnings`, it outputs the following:

```
(node:22356) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 2 SIGWINCH listeners added to [process]. Use emitter.setMaxListeners() to increase limit
    at _addListener (events.js:390:17)
    at process.addListener (events.js:406:10)
    at process.getStderr [as stderr] (internal/bootstrap/switches/is_main_thread.js:135:13)
    at ReadStream.Readable.pipe (_stream_readable.js:656:32)
    at C:\projects\lightweight-charts\node_modules\@size-limit\file\index.js:37:42
    at new Promise (<anonymous>)
    at gzipSize (C:\projects\lightweight-charts\node_modules\@size-limit\file\index.js:35:10)
    at C:\projects\lightweight-charts\node_modules\@size-limit\file\index.js:61:48
    at Array.map (<anonymous>)
    at sum (C:\projects\lightweight-charts\node_modules\@size-limit\file\index.js:11:35)
```

Similar to #146.

Both `size-limit` and `@size-limit/file` are `4.5.2` version.

**Comment by timocov** [comment]
It's interesting that running `size-limit` doesn't cause this warning (with enabled `file` check) 🤔

**Comment by timocov** [comment]
@ai from https://github.com/ai/size-limit/issues/146#issuecomment-651444680:

> Start with Size Limit CLI. Maybe it is a problem of JS APII(which is not really user-friendly)

We use both CLI and JS API, because we need to be able to check that size isn't out of our limits and need to see what's the difference since merge-base. I saw GitHub Action for that, but we can't use GitHub Actions for some reasons, unfortunately - that's why we write in-house script to handle this.

> If you still will have a problem, open another issue

Yep, here we are :)

**Comment by ai** [comment]
Am I right that you have this issue only in JS API?

**Comment by timocov** [comment]
Yes, that's correct.

**Comment by ai** [comment]
Show your Size Limit CLI config

**Comment by timocov** [comment]
You don';t need to have it at all (since you don't run CLI which uses that config). I created a repro of the issue if you'd like:

1. Download and unzip [size-limit-test.zip](https://github.com/ai/size-limit/files/4850076/size-limit-test.zip)
2. Run `npm install`
3. Run `node run-size-limit.js`

**Comment by ai** [comment]
Thanks. I will look tomorrow.

**Comment by ai** [comment]
This issue is related with these fixes:
https://github.com/ai/size-limit/pull/150
https://github.com/ai/size-limit/pull/149

@pustovalov do you an idea why fixes stopped to work?

**Comment by timocov** [comment]
@ai @pustovalov When I tried to "fix" it from the script, just setting `process.setMaxListeners(100)` didn't work for me. I figured out that it's insufficiently to just increase max listeners, it's needed to write something to stdout (I don't know why), so my "final fix" (which helps from the script) is something like this:

```javascript
function fixMaxListenersExceededWarning() {
	process.setMaxListeners(100);

	process.stderr.getMaxListeners();
	// DON'T REMOVE IT!
	// WTF Node?! ¯\_(ツ)_/¯
	process.stdout.write('');
}
```

Maybe this helps :)

**Comment by pustovalov** [comment]
Hey @ai, looks like when user use JS API, there is another additional process

```
[~/Downloads/size-limit-test]$ node run-size-limit.js                                                                                                                                  [ruby-head]
checks [ { path: [ 'package.json' ] } ]
count of checks 1
(node:58545) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 2 SIGWINCH listeners added to [process]. Use emitter.setMaxListeners() to increase limit
(Use `node --trace-warnings ...` to show where the warning was created)
[ { size: 201 } ]
```

If I do this one: `process.setMaxListeners(2)` works fine for me, without `process.stdout.write('');`

```
[~/Downloads/size-limit-test]$ node run-size-limit.js                                                                                                                                  [ruby-head]
checks [ { path: [ 'package.json' ] } ]
count of checks 2
[ { size: 201 } ]
```

**Comment by ai** [comment]
Fixed by https://github.com/ai/size-limit/pull/187

**Comment by ai** [comment]
The fix was released in 4.5.3.

**Comment by timocov** [comment]
Thank you!
